### PR TITLE
php: 7.2.11 -> 7.2.12, 7.1.23 -> 7.1.24 (CVE-2018-17082)

### PR DESCRIPTION
**Pull request by delroth**
Also align Darwin with Linux versions by introducing some trivial configure.ac
patches. This requires running autoconf/autoheader (through ./buildconf), which
also made a change to fix-paths-php7.patch necessary since it patched ./configure
directly instead of the .m4 inputs.

Supersedes #50511 with backwards compatibility on Darwin. @Ekleog @etu 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by samueldr** [comment]
@GrahamcOfBorg build php71 php72

@GrahamcOfBorg test elk owncloud

**Comment by Ekleog** [comment]
@grahamcofborg build nixosTests.elk nixosTests.owncload

**Comment by Ekleog** [comment]
@GrahamcOfBorg build nixosTests.elk nixosTests.owncloud

Weird the build passed with a typo like this.

**Comment by Ekleog** [comment]
@delroth Could you open a PR with backports of these commits to `release-18.09`? Using `git cherry-pick -x` :)

File: pkgs/development/interpreters/php/default.nix [hunk 1]
```diff
@@ -223,35 +226,19 @@ let
...
-  # We can't update the darwin versions because they simply don't compile at
-  # all due to a bug in the intl extensions.
-  #
-  # The bug so far is present in 7.1.21, 7.1.22, 7.1.23, 7.2.9, 7.2.10, 7.2.11.
+  php71 = generic {
+    version = "7.1.23";
+    sha256 = "0jyc5q666xh808sgy78cfylkhy5ma2zdg88jlxhagyphv23aly9d";
+
+    # https://bugs.php.net/bug.php?id=76826
+    extraPatches = optional stdenv.isDarwin [ ./php71-darwin-isfinite.patch ];
```
Comment (Ekleog) [inline pkgs/development/interpreters/php/default.nix]: The `[]` brackets are useless here, they're already added by `optional`, so `patches` ends up containing `[ ./fix-paths-php7.patch [ ./php71-darwin-isfinite.patch ] ]`, which I guess nix handles correctly, but… :)

File: pkgs/development/interpreters/php/default.nix [hunk 2]
```diff
@@ -223,35 +226,19 @@ let
...
-        sha256 = "1rky321gcvjm0npbfd4bznh36an0y14viqcvn4yzy3x643sni00z";
-      }
-    else
-      {
-        version = "7.2.11";
-        sha256 = "1idlv04j1l2d0bn5nvfrapcpjh6ayj1n4y80lqvnp5h75m07y3aa";
-      }
-  );
+    # https://bugs.php.net/bug.php?id=76826
+    extraPatches = optional stdenv.isDarwin [ ./php72-darwin-isfinite.patch ];
```
Comment (Ekleog) [inline pkgs/development/interpreters/php/default.nix]: Same here

**Review by Ekleog — COMMENTED** [review]
Sorry for not noticing this before asking you to backport.

**Comment by Ekleog** [comment]
OK so `php71` and `php72` passed everywhere but on darwin, where it failed due to a timeout due to too many rebuilds. `owncloud` and `elk` are still pending on `aarch64-linux` and `x86_64-linux`.

Let's retry the darwin build, as IIRC there is only one darwin builder it should re-use the now-cached builds, and it's idling currently anyway.

@grahamcofborg build php71 php72

**Comment by Ekleog** [comment]
O~kay, it looks like the darwin build of clang has been restarted on hydra (it had been cancelled) and passed, so hopefully this time ofborg won't timeout on darwin :)

@GrahamcOfBorg build php71 php72

**Review by etu — APPROVED** [review]

**Comment by Ekleog** [comment]
Looks like it built correctly, thank you!

**Pull request merged**